### PR TITLE
Add basic reprojection documentation

### PR DESCRIPTION
**Pull request by keflavich**
Still something of a work-in-progress as I haven't tested the WCS tools nor gotten all the .rst syntax right.

cc @e-koch

### CODE REVIEW

**Comment by keflavich** [comment]
OK I think all the links work now

File: docs/reprojection.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+<https://github.com/radio-astro-tools/tutorials/blob/master/SpectralCubeReprojectExample.ipynb>`_.
+
+
+Spatial Reprojection
+^^^^^^^^^^^^^^^^^^^^
+
+To reproject a cube onto a different spatial world coordinate system, use the
+:meth:`~spectral_cube.SpectralCube.reproject` function.  The function requires
+a target header as an input.  You might generate this header by grabbing it
+from another FITS file, generating it with a helper tool (e.g., the
```
Comment (e-koch) [inline docs/reprojection.rst:22]:
```suggestion
from another FITS file, for example, from another `SpectralCube`::

    from spectral_cube import SpectralCube
    
    cube = SpectralCube.read('/some_path/some_file.fits')
    other_cube = SpectralCube.read('/some_path/other_file.fits')

    reprojected_cube = cube.reproject(other_cube.header)

Instead, the target header can be generated it with a helper tool (e.g., the
```

Comment (e-koch) [reply docs/reprojection.rst:22]: Add another common use case.

File: docs/reprojection.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+`find_optimal_celestial_wcs
+<https://reproject.readthedocs.io/en/stable/mosaicking.html#computing-an-optimal-wcs>`_
+function in the `reproject <https://reproject.readthedocs.io/>`_ package), or
+manually editing a FITS header.
```
Comment (e-koch) [inline docs/reprojection.rst:26]:
```suggestion
by manually editing a FITS header.
```

File: docs/reprojection.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+The spatial reprojection tool uses reproject_ under the hood and defaults to
+using a bilinear interpolation scheme, though this is configurable.
+Interpolation onto a differently-spaced grid, after appropriate :doc:`smoothing <smoothing>`, can
+be used to rebin or decimate the data.
+
+A simple example for rebinning, assuming no smoothing is needed (the data are oversampled)::
```
Comment (e-koch) [inline docs/reprojection.rst:33]:
```suggestion
A simple example for rebinning, assuming no smoothing is needed (appropriate for when the data are oversampled)::
```

File: docs/reprojection.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    cube = SpectralCube.read('/some_path/some_file.fits')
+
+    # create a target header to reproject to
+    target_header = cube.wcs.celestial[::2, ::2].to_header()
+    target_header['NAXIS1'] = cube.shape[2] / 2
+    target_header['NAXIS2'] = cube.shape[1] / 2
+
+    downsampled_cube = cube.reproject(target_header)
+
+
```
Comment (e-koch) [inline docs/reprojection.rst:47]:
```suggestion
Reprojection for 2D images uses the same syntax with a `~spectral_cube.Projection` or `~spectral_cube.Slice` object. For example, to match the spatial grid of a 2D image to that of a cube::

        from spectral_cube import SpectralCube, Projection
        from astropy.io import fits
    
        cube = SpectralCube.read('/some_path/some_file.fits')
        image = Projection.from_hdu(fits.open('/some_path/twod_image.fits')[0])
        
        cube_header_spatial = cube.wcs.celestial.to_header()
        reprojected_image = image.reproject(cube_header_spatial)

```
2D example.

File: docs/reprojection.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+be used to rebin or decimate the data.
+
+A simple example for rebinning, assuming no smoothing is needed (the data are oversampled)::
+
+
+    from spectral_cube import SpectralCube
+
+    cube = SpectralCube.read('/some_path/some_file.fits')
+
+    # create a target header to reproject to
```
Comment (e-koch) [inline docs/reprojection.rst:40]:
```suggestion
    # create a target header to reproject to by making the pixel size 2 times larger
```

File: docs/reprojection.rst [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+:meth:`~spectral_cube.SpectralCube.reproject` function.  The function requires
+a target header as an input.  You might generate this header by grabbing it
+from another FITS file, generating it with a helper tool (e.g., the
+`find_optimal_celestial_wcs
+<https://reproject.readthedocs.io/en/stable/mosaicking.html#computing-an-optimal-wcs>`_
+function in the `reproject <https://reproject.readthedocs.io/>`_ package), or
+manually editing a FITS header.
+
+The spatial reprojection tool uses reproject_ under the hood and defaults to
+using a bilinear interpolation scheme, though this is configurable.
```
Comment (e-koch) [inline docs/reprojection.rst:29]:
```suggestion
using a bilinear interpolation scheme, though this is configurable using the
`order` keyword in `~spectral_cube.SpectralCube.reproject`.
```

**Review by e-koch — COMMENTED** [review]
Minor comments + a couple example additions I'll throw back to you @keflavich to look over.

Otherwise, looks good. Links work and I don't see issues in the docs build.

**Pull request merged**